sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingSessionWindowSuite.scala

```diff
@@ -43,12 +43,17 @@ class StreamingSessionWindowSuite extends StreamTest
     val mergingSessionOptions = Seq(true, false).map { value =>
       (SQLConf.STREAMING_SESSION_WINDOW_MERGE_SESSIONS_IN_LOCAL_PARTITION.key, value)
     }
-    val providerOptions = Seq(
+    var providerOptions = Seq(
       classOf[HDFSBackedStateStoreProvider].getCanonicalName,
       classOf[RocksDBStateStoreProvider].getCanonicalName
     ).map { value =>
       (SQLConf.STATE_STORE_PROVIDER_CLASS.key, value.stripSuffix("$"))
     }
+    // RocksDB doesn't support Apple Silicon yet
+    if (System.getProperty("os.name").equals("Mac OS X") &&
+        System.getProperty("os.arch").equals("aarch64")) {
+      providerOptions = providerOptions.take(1)
```

Comment:
nit but maybe it would be better to explicitly filter `RocksDBStateStoreProvider` out

Reply:
Sure.